### PR TITLE
Add more verbose error messages

### PR DESCRIPTION
**Pull request by alexmasi**
Also add watching for namespace deletion to avoid cases errors such as:

```
$ ./kne_cli create ../examples/openconfig/lemming.pb.txt
I0802 19:08:06.480612  630769 root.go:143] /usr/local/google/home/alexmasi/kne/examples/openconfig
I0802 19:08:06.482492  630769 topo.go:161] Trying in-cluster configuration
I0802 19:08:06.482515  630769 topo.go:164] Falling back to kubeconfig: "/usr/local/google/home/alexmasi/.kube/config"
I0802 19:08:06.483907  630769 topo.go:421] Adding Link: lemming1:eth1 lemming2:eth1
I0802 19:08:06.483929  630769 topo.go:462] Adding Node: lemming1:OPENCONFIG
I0802 19:08:06.483951  630769 topo.go:462] Adding Node: lemming2:OPENCONFIG
Error: failed to create topology "lemming-twodut": failed to create meshnet topologies: could not create topology for meshnet node lemming1: topologies.networkop.co.uk "lemming1" is forbidden: unable to create new content in namespace lemming-twodut because it is being terminated
```

when using `kne create` right after `kne delete`

### CODE REVIEW

**Comment by alexmasi** [comment]
example:

```
$ ./kne_cli deploy ../../kne-internal/deploy/kne/kind-bridge.yaml
I0801 21:45:24.474877  347486 deploy.go:248] Deploying cluster...
I0801 21:45:24.488880  347486 deploy.go:534] kind version valid: got v0.19.0 want v0.17.0
I0801 21:45:24.488968  347486 deploy.go:570] Creating kind cluster with: [create cluster --name kne --image us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test --config /usr/local/google/home/alexmasi/kne-internal/manifests/kind/config.yaml]
W0801 21:45:24.669572  347486 deploy.go:84] (kind): Creating cluster "kne" ...
W0801 21:45:24.669910  347486 deploy.go:84] (kind):  • Ensuring node image (us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test) 🖼  ...
W0801 21:45:39.235551  347486 deploy.go:84] (kind):  ✗ Ensuring node image (us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test) 🖼
W0801 21:45:39.268902  347486 deploy.go:84] (kind): ERROR: failed to create cluster: failed to pull image "us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test": command "docker pull us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test" failed with error: exit status 1
W0801 21:45:39.268929  347486 deploy.go:84] (kind): Command Output: Error response from daemon: manifest for us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test not found: manifest unknown: Failed to fetch "test"
Error: failed to deploy cluster: failed to create kind cluster: "/usr/local/google/home/alexmasi/go/bin/kind create cluster --name kne --image us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test --config /usr/local/google/home/alexmasi/kne-internal/manifests/kind/config.yaml" failed: exit status 1: ERROR: failed to create cluster: failed to pull image "us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test": command "docker pull us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test" failed with error: exit status 1, Command Output: Error response from daemon: manifest for us-west1-docker.pkg.dev/gep-kne/kne/kindest/node:test not found: manifest unknown: Failed to fetch "test"
Log files can be found in:
    /tmp/kne_cli.alexmasi.alexmasi.log.INFO.20230801-214524.347486
    /tmp/kne_cli.alexmasi.alexmasi.log.WARNING.20230801-214524.347486
```

File: deploy/deploy.go [hunk 1]
```diff
@@ -220,25 +243,25 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) (rerr error) {
...
 	}
 	log.Infof("Deploying cluster...")
 	if err := d.Cluster.Deploy(ctx); err != nil {
-		return err
+		return fmt.Errorf("failed to deploy cluster: %w", err)
 	}
 	log.Infof("Cluster deployed")
 	if err := d.Cluster.Healthy(); err != nil {
-		return err
+		return fmt.Errorf("failed to check cluster is healthy: %w", err)
```
Comment (NehaManjunath) [inline deploy/deploy.go:254]: nit: "failed to check if the cluster is healthy: %w"

File: deploy/deploy.go [hunk 2]
```diff
@@ -298,35 +321,35 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) (rerr error) {
...
 	log.Infof("Deploying ingress...")
 	if err := d.Ingress.Deploy(ctx); err != nil {
-		return err
+		return fmt.Errorf("failed to deploy ingress: %w", err)
 	}
 	tCtx, cancel := context.WithTimeout(ctx, healthTimeout)
 	defer cancel()
 	if err := d.Ingress.Healthy(tCtx); err != nil {
-		return err
+		return fmt.Errorf("failed to check ingress is healthy: %w", err)
```
Comment (NehaManjunath) [inline deploy/deploy.go:329]: same as above

File: deploy/deploy.go [hunk 3]
```diff
@@ -298,35 +321,35 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) (rerr error) {
...
 	if err := d.CNI.Deploy(ctx); err != nil {
-		return err
+		return fmt.Errorf("failed to deploy CNI: %w", err)
 	}
 	d.CNI.SetKClient(kClient)
 	tCtx, cancel = context.WithTimeout(ctx, healthTimeout)
 	defer cancel()
 	if err := d.CNI.Healthy(tCtx); err != nil {
-		return err
+		return fmt.Errorf("failed to check CNI is healthy: %w", err)
```
Comment (NehaManjunath) [inline deploy/deploy.go:340]: same as above

File: deploy/deploy.go [hunk 4]
```diff
@@ -298,35 +321,35 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) (rerr error) {
...
 		if err := c.Deploy(ctx); err != nil {
-			return err
+			return fmt.Errorf("failed to deploy controller: %w", err)
 		}
 		c.SetKClient(kClient)
 		tCtx, cancel = context.WithTimeout(ctx, healthTimeout)
 		defer cancel()
 		if err := c.Healthy(tCtx); err != nil {
-			return err
+			return fmt.Errorf("failed to check controller is healthy: %w", err)
```
Comment (NehaManjunath) [inline deploy/deploy.go:352]: same

File: deploy/deploy.go [hunk 5]
```diff
@@ -71,6 +72,28 @@ func logCommand(cmd string, args ...string) error {
 	return c.Run()
 }
 
+// outLogCommand runs the specified command but records standard output
+// with log.Info and standard error with log.Warning. Standard output
```
Comment (NehaManjunath) [inline deploy/deploy.go:76]: Since standard errors are being recorded, should the log level be log.Error instead of Warning?

Comment (alexmasi) [reply deploy/deploy.go:76]: Sometimes CLIs write to stderr even when there is not actually an error that we would need to concern KNE users with. If the cmd fails the user should get an error message from KNE with the relevant bits of the stderr/out included. I believe that warning is more appropriate in this case

Comment (NehaManjunath) [reply deploy/deploy.go:76]: Thanks for the clarification!

**Review by NehaManjunath — APPROVED** [review]

**Pull request merged**